src/index.js

```diff
@@ -117,13 +117,15 @@ module.exports = exports = options => {
   async function broadcast(data, options, clients) {
     options = options || {}
     const rms = options.rooms || []
-    if (rms.length === 0) {
+    let ids = []
```

Comment:
I was thinking at a different approach. Instead of creating an array with all the ids an then removing the duplicates, use a set and the pass that set to the `send()` function.
In `send()` use the set to see if the message has already been send to the socket.

So basically replacing this line with `const sent = new Set();` and in `send()`:

```js
function send(sent, ids, clients, data, options) {
  ...
  ids.forEach(id => {
    const socket = clients[id]
    if (!socket || ~except.indexOf(id) || sent.has(id)) return
    sent.add(id);
    transform(socket, data, method, transformer)
  })
}
```

Makes sense? Thoughts?